### ISSUE TITLE
Building static site with v4.2.1 seems to be broken

### ISSUE THREAD

**Issue by ohmykreee** [Lednerb/bilberry-hugo-theme#613]
Theme version I used: v4.2.1

Hugo version I used: v0.121.1 (Local machine and GitHub Actions)

## Issues:
I imported this theme as module with Option 1:
```toml
# Option 1 (recommended): adding the theme as a Hugo module
theme = "github.com/Lednerb/bilberry-hugo-theme/v4"
```
Everything works fine when using `hugo server` and `hugo server -e production`. But when using `hugo` or `hugo --minify` to generate static file and host it to http server, the style is totally broken, with errors in the developer tools:
[1f66763a.blog-7jo.pages.dev-1702384047429.log](https://github.com/Lednerb/bilberry-hugo-theme/files/13648052/1f66763a.blog-7jo.pages.dev-1702384047429.log)

I've also tested on my machine, with first executed `hugo mod clean --all` to clean all module, then `hugo mod get -u` to update module, but the results were the same: `hugo server` works but `hugo` failed.

---

You can refer my GitHub Actions run here (with build result in the artifacts): https://github.com/ohmykreee/kreee-blog/actions/runs/7176741651

And preview the results here: https://1f66763a.blog-7jo.pages.dev/

All my codes concerning theme v4 are on branch `migrate-theme-v4`: https://github.com/ohmykreee/kreee-blog/tree/migrate-theme-v4

If anything you need please let me know.

Thanks in advance.

**Comment by igor-baiborodine** [comment]
@ohmykreee, try modifying this [line](https://github.com/ohmykreee/kreee-blog/blob/68e953fa00b161b57e7f0e9d4e315a44ee8fec6e/.github/workflows/publish-site.yml#L26) as follows:

run: |
    hugo mod get
    hugo --minify

I guess it's broken since the Bilberry theme's Hugo module is not imported before executing the `hugo --minify` command.

**Comment by ohmykreee** [comment]
> @ohmykreee, try modifying this [line](https://github.com/ohmykreee/kreee-blog/blob/68e953fa00b161b57e7f0e9d4e315a44ee8fec6e/.github/workflows/publish-site.yml#L26) as follows:
> 
> run: | hugo mod get hugo --minify
> 
> I guess it's broken since the Bilberry theme's Hugo module is not imported before executing the `hugo --minify` command.

The issue still exists with new line added in the workflow file.

Here is the new actions run: https://github.com/ohmykreee/kreee-blog/actions/runs/7182515477

And the preview link: https://9f6654c6.blog-7jo.pages.dev/

**Comment by igor-baiborodine** [comment]
@ohmykreee, it's not working since it's pulling the module for `v1.0.1`:

![image](https://github.com/Lednerb/bilberry-hugo-theme/assets/1027701/0804b546-f99c-4b86-ba06-1f42a8aec1c2)

Try adding the `go.sum` file to your branch. Then the `hugo mod get` will be redundant.
Here is an example: https://github.com/igor-baiborodine/kiroule.com/blob/master/go.sum

**Comment by ohmykreee** [comment]
It appears that I messed up the Go module version management, and I can't locate where the mistake is. So I've now switched to using Git submodules for importing the theme, which have already resolved this problem.

Additionally, I identified the source of the 404 errors for the missing JavaScript files. The links were referencing the `baseURL` configuration, which was omitted during the migration from v3 to v4. This has now been corrected.

Thank you for your patience and assistance in resolving these issues.
